test/functional/tests/memory/test_memory_metadata_consumption.py

```diff
@@ -0,0 +1,191 @@
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import ReadWrite, IoEngine
+from test_utils.os_utils import get_mem_available, get_module_mem_footprint
+from test_utils.os_utils import sync, Udev
+from test_utils.size import Size, Unit
+
+cores_count = 16
+cache_mode_wb = CacheMode.WB
+percentage_tolerance = 5
+cas_cache_module = 'cas_cache'
+
+
+@pytest.mark.parametrizex("cache_mode", CacheMode)
+@pytest.mark.parametrizex("cache_line_size", CacheLineSize)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_memory_metadata_consumption_empty_cache(cache_line_size, cache_mode):
+    """
+        title: Test for CAS RAM consumption for metadata.
+        description: |
+          Validate CAS RAM consumption for metadata depends on start command parameters.
+        pass_criteria:
+          - Successful CAS creation and cores addition.
+          - Memory consumption within limits.
+    """
+    with TestRun.step("Prepare drives and partitions for core devices."):
+        cache_disk = TestRun.disks['cache']
+        cache_disk.create_partitions([Size(50, Unit.GibiByte)])
+        cache_dev = cache_disk.partitions[0]
+
+        core_disk = TestRun.disks['core']
+        parts = [Size(3200, Unit.MebiByte)] * cores_count
+        core_disk.create_partitions(parts)
+        core_devices = core_disk.partitions
+
+    with TestRun.step("Start cache with configuration and add all core devices."):
+        free_mem_before_start = get_mem_available()
+
+        cache = casadm.start_cache(cache_dev, cache_mode, cache_line_size, force=True)
+
+        for dev in core_devices:
+            cache.add_core(dev)
+
+    with TestRun.step("Measure allocates size of metadata."):
+        used_memory = get_module_mem_footprint(cas_cache_module)
+        TestRun.LOGGER.info(f"Memory allocated: {used_memory}")
+
+    with TestRun.step("Compare allocated size with cache usage statistics "
+                      "and required consumption."):
+        free_mem_after_start = get_mem_available()
+
+        metadata_max_size = Cache.calculate_metadata_max_size(cache.size, cache_line_size)
+        TestRun.LOGGER.info(f"Metadata max size: {metadata_max_size}")
+
+        validate_memory_consumption(cache, metadata_max_size, used_memory)
+
+        memory_delta = free_mem_before_start - free_mem_after_start
+        ram_change_status = is_memory_within_limit("Measured increment of RAM usage after start",
+                                                   metadata_max_size, memory_delta)
+
+        TestRun.LOGGER.info("Stop CAS and verify memory changes again")
+        casadm.stop_all_caches()
+
+        free_mem_after_stop = get_mem_available()
+        memory_delta = free_mem_after_stop - free_mem_after_start
+        ram_change_status |= is_memory_within_limit("Measured decrement of RAM usage after stop",
```

Comment:
I'm not sure about this check. It would pass if memory usage did not drop after stopping CAS. On other hand it would fail if fore some reason there is slightly more memory available in the system for some unrelated reason once CAS is stopped. I would remove this check.